### DISCUSSION TITLE
Why are the screen states conform to Codable in the NavigationStack case study?

### DISCUSSION THREAD

**Discussion by mycroftcanner** [pointfreeco/swift-composable-architecture#2537] [Q&A]
While diving into the NavigationStack case study, I noticed that screen states are expected to comply with `Codable`. Could someone enlighten me on the rationale behind this decision? Thanks in advance

**Comment by mbrandonw** [comment]
Hi @mycroftcanner, features are not expected or required to have `Codable` state to be used in a navigation stack. The use of `Codable` in the navigation stack case study could be dropped and it would still compile. I think we just had that left over from some early experiments.

If you wanted to PR that removal we would happily merge it.
